pkg/builder/statefulset_container_builder.go

```diff
@@ -33,6 +33,19 @@ func (b *Builder) buildStsContainers(mariadb *mariadbv1alpha1.MariaDB, dsn *core
 		}
 		containers = append(containers, buildMetricsContainer(mariadb.Spec.Metrics, dsn))
 	}
+	if mariadb.Spec.SidecarContainers != nil {
+		for index, containerTpl := range mariadb.Spec.InitContainers {
+			sidecarContainer := buildContainer(&containerTpl)
+			sidecarContainer.Name = fmt.Sprintf("init-%d", index)
```

Comment:
```suggestion
			sidecarContainer.Name = fmt.Sprintf("sidecar-%d", index)
```

Reply:
This is to avoid clashing with the `initContainers`

Reply:
Agree, I overlooked it, thank you.